### ISSUE TITLE
Why does JSMA use probs in the interface?

### ISSUE THREAD

**Issue by goodfeli** [tensorflow/cleverhans#853]
JSMA is one of the last attacks that still uses probs in the interface, due to the risk of gradient masking from loss of precision in the softmax. I'm not very familiar with how this attack works so I'm not sure whether we should rewrite it to use logits in the interface or whether we should update the doc to make it clear why probs are safe / necessary.

**Comment by npapernot** [comment]
(Discussed offline) changing it to use the logits should not be a problem
